lib/buildreq.c

```diff
@@ -175,7 +175,7 @@ int rc_aaa_ctx_server(rc_handle * rh, RC_AAA_CTX ** ctx, SERVER * aaaserver,
 
 		result = rc_send_server_ctx(rh, ctx, &data, msg, type);
 
-		if ((result == OK_RC) || (result == CHALLENGE_RC)) {
```

Comment:
Hi,
I have a few clarifications. 

Hi @martin-belanger ,
>>>"On a REJECT we currently move to the next server in the list (if any). "

On a REJECT scenario, this REJECT info has to be passed on to the Caller(Client) as the Server has rejected the respective credentials. IMHO, the received REJECT message has to be processed by the caller. And even if we move on to next RADIUS Server, most probably a REJECT would be received again due  to invalid credentials of caller. I beleive, onlyincase of TIME_OUT, moveing onto the next RADIUS Server would greatly help. 

>>> "on a REJECT with EAP contents and if the flag RC_FLAG_AVPAIR_ON_REJECT is set, then return."

The REJECT message may contain not only EAP contents but also other RADIUS Attributes that are designed by Admin to place the users with invalid credentials to a certain administrative domain like Vlan and so on. Hence IMHO, checking for EAP contents alone may not be sufficient. 

Hi @nmav , 
>>> "That would cause a memory leak on existing applications which receive a reject. "

If possible, kindly let me know the message flow where the memory leak would happen. I think, the REJECT msg should be processed same as ACCEPT msg and Caller should be able to process all the attributes received from Server from REJECT scenario too. And similar to ACCEPT, the AV_pair list should be mem-freed before proceeding to any further action. 

Hi @nmav , @martin-belanger ,

Kindly let me know your views.